static/game/powerups/xplode.js

```diff
@@ -12,41 +15,83 @@ export class Xplode extends PowerUp {
    */
   constructor(pos, magnitude = 1) {
     super(pos, magnitude, "Xplode");
-    this.powerUpName = this.powerUpClass + " " + this.magnitude;
   }
 
   /**
    * applies this powerup
    * @param {Creature} creature
    * @override
-   * @returns {Boolean}
    */
   apply(creature) {
-    if (!super.apply(creature)) {
+    if (!this.isAtMax(creature)) {
+      super.apply(creature);
+      /**
+       * @param {Bullet} b the parent bullet
+       * @param {number} [num] the number of bullets to spawn
+       */
+      const f = (b, num = 1) => {
+        /** @type {number} */
+        const mag = b.vel.mag();
+        let theta = Math.random() * 2 * Math.PI;
+        for (let i = 0; i < num; i++) {
+          // rotate around so new bullets are distributed evenly
+          theta += (1 / num) * 2 * Math.PI;
+          const newVel = new Vector(Math.cos(theta), Math.sin(theta)).norm2();
+          const child = creature.getBullet(newVel, b.good, b.color);
+          child.vel = child.vel.norm2().mult(creature.bulletSpeed * 0.75);
+          child.pos = b.pos;
+          /**
+           * remove xplode functions from child so it doesn't multiply forever
+           * @type {
+           *   {name: string, data: number, func: (function(Bullet): void)}[]
+           * }
+           */
+          const newOnDestroy = new Array();
+          for (const od of b.onDestroy) {
+            if (od.name && od.name !== this.powerUpClass) {
+              newOnDestroy.push(od);
+            }
+          }
+          child.onDestroy = newOnDestroy;
+          addToWorld(child);
+        }
+      };
+
+      creature.fireDelay += FIRE_DELAY_ADDEND * this.magnitude;
```

Comment:
So if we get a level 5 Xplode it adds 50 to our fire delay? If you're at max fire rate, that means your DPS will be divided by like 10, even taking into account the extra bullets that spawn. Maybe it can scale then floor your fire rate instead of having an addend? Of course we'll probably just have to play it and see what works.

Reply:
I played around with this a bit and couldn't find a good compromise. We can always balance it more later when we get the spawning probabilities in